### PR TITLE
fixing https://github.com/http4s/http4s/issues/334

### PR DESCRIPTION
**Pull request by ShengC**

### CODE REVIEW

File: blaze-core/src/main/scala/org/http4s/blaze/websocket/Http4sWSStage.scala [hunk 1]
```diff
@@ -90,24 +81,24 @@ class Http4sWSStage(ws: ws4s.Websocket) extends TailStage[WebSocketFrame] {
         logger.trace(t)("WebSocket Exception")
         sendOutboundCommand(Command.Disconnect)
     }
-
-    ws.exchange.read.through(sink).run.runAsync(onFinish)
+    
+    (alive.discrete.map(!_)).wye(ws.exchange.read.to(snk))(wye.interrupt).run.runAsync(onFinish)
```
Comment (bryce-anderson) [inline blaze-core/src/main/scala/org/http4s/blaze/websocket/Http4sWSStage.scala]: Could we flip the boolean meaning from `alive` to `dead` or something so we don't need to do a `map` here? It seems wasteful to need to create all those objects just to do a negation.

**Comment by bryce-anderson** [comment]
This looks good to me, although I wonder if we will have the same problems on the input side. This could certainly use some tests to prevent regressions, but I won't let that stop this from getting merged.
:+1:

**Comment by bryce-anderson** [comment]
Thanks @ShengC, I'll merge this by the end of the day barring objections from others.

**Comment by ShengC** [comment]
Wondering when this can be merged and released ? I am trying to use http4s in one of my projects due to its elegant way of dealing with websocket, but this bug is definitely a blocker for anything I want to build upon websocket.

**Comment by ShengC** [comment]
@bryce-anderson Just saw your comments - thanks ! Do you have an idea when this could be released after being merged? It is currently a blocker for one of my projects.

**Comment by bryce-anderson** [comment]
@ShengC, I'd like to get #338 into a 0.8.4 bug fix release. I expect that to happen before the end of the weekend.

**Comment by bryce-anderson** [comment]
@ShengC, thanks for the PR! :smile:

**Comment by rossabaker** [comment]
Note that there is already a branch for said release.

**Comment by bryce-anderson** [comment]
Oh my, I didn't see that branch.

**Pull request merged**